### ISSUE TITLE
Remote working - how to do it efficiently 

### ISSUE THREAD

**Issue by iteles** [nelsonic/remote-working-experiment#8]
@naazy As you're our remote working 'experiment' :wink:, it would be awesome if whenever you feel something is working well for you, you could drop a note on it here and whenever you feel something is not, do the same. What do you think?

@Conorc1000 and @roryc89, as the other half of the team, it would be great to get your thoughts too?

cc/ @Jbarget, you might be interested in this 😊

**Comment by naazy** [comment]
Things working well/Positive observations
- Making sure you and the team assign yourself to issues and mark 'in progress'
- Extra pressure to be productive because you're out of sight

Things that could work better/General advice
- If you're working from a cafe, make sure that the internet is reliable and doesn't have a download limit! Also, I've found it weirdly hard to find cafes with plugs!
- Try to find a fallback video conferencing service (for when Google Hangouts inevitably breaks!)

**Comment by Jbarget** [comment]
One thing i always do now, even if the cafe has a sign saying they have Wifi, to ask if its working cos previously i've bought something and the wifi wasnt working that day

**Comment by naazy** [comment]
Yeah I had the same thing earlier today but luckily realised it wasn't working before I'd bought anything

**Comment by iteles** [comment]
Loving the advice so far <3

**Comment by Jbarget** [comment]
After a recent trip away to Sweden I found:
- being with someone who wasn't remote working/having a similar day structure made it harder to get into a routine
- when the person I was with had spare time, I felt pressure to spend that spare time with them (also because I was visiting for 5 days)
- waking up early (8am) to start my day was beneficial (but harder to do if you're having social evenings)

In future I think it would be best, especially since I am 'untrained' as remote a worker:
- to set myself a timetable of the work I aim to do, rather than saying generically 'i'll do the work'
- work time is work time, play time is play time
- inform anyone I am there with about my timetable so they understand that I need to work
-  let them know when they make comments that make me feel pressured to do something with them (a polite 'You do you, I'll do me' will probably suffice @naazy)

**Comment by nelsonic** [comment]
@jrans how would you have improved your remote working while on "ART" ?
@jackcarlisle you've had a bit less experience of working remotely, but I _suspect_ it's something you _might_ want to do a bit more of in the future ... 

Specifically: 
- when were you most _productive_/_effective_?
- how would you rate your team's communication? (_was anything ever miss-communicated/lost_)
- what could have been improved?
- did you follow a specific problem-solving approach?
- what was the hardest problem you had to solve? (_share an example!_)

**Comment by jackcarlisle** [comment]
- most productive when I work somewhere quiet with good wifi for example the Wellcome Collection Library. Also more productive when I have the appropriate setup for work i.e. good desk, comfortable chair and another monitor (_if possible_). I've found that I work better if I get out of the house and go to a different location. The change of scenery helps me focus.
- team on previous project communicated well at the beginning but when more people got added to the project at different times the processes for sharing knowledge was slightly skewed.
- possible improvements could have included more code reviews across the team so that you gain more exposure to the whole codebase if working in more rigid roles and also more structure when working on new features
- usually I start by writing out the problem in plain english on a piece of paper. I then break it down into smaller sub tasks and then map out how I think they should be solved in technical terms. I then start with the first one and test out some of the functionality in jsbin to make sure it works in the way I expect. I then add more pieces together checking that everything works the way it should until the problem is solved. After that I refactor where necessary.
- hard to think but one of the more recent difficult problems was the [S3 Direct upload from the browser](https://github.com/dwyl/image-uploads/tree/master/examples/direct-upload) because of the fact that there wasn't a lot of good documentation and a good portion of it was out of date.

Hope this helps!

**Comment by Conorc1000** [comment]
My main observation while remote working has been that It becomes much easier as the project becomes more mature, and can even lead to better documentation. 
I think if you tried to remote work for the first couple of sprints you would miss out on lots of information just from chatting to / over hearing you colleages.

**Comment by Jbarget** [comment]
GitLab is a mostly remote working company that likes the open-sauce

See below for their blog posts on the matter:

https://about.gitlab.com/2015/04/08/the-remote-manifesto/
https://about.gitlab.com/2016/12/05/how-we-stay-connected-as-a-remote-company/
https://about.gitlab.com/2016/03/04/remote-working-gitlab/

PS. they also have a handbook for their company as a whole:
https://about.gitlab.com/handbook/

**Comment by iteles** [comment]
@Conorc1000 Particularly like that you mentioned 'overhearing' - I've found that to be by far the most unexpected benefit of co-location :+1:

**Comment by franzmoro** [comment]
I am generally very happy about the remote experience: I think that learning the practices that "make remote-working work",  have led me to getting better at both individual and team work. I thus find that most of the following tips are also applicable for non-remote working (although probably less mandatory if co-located). I'll keep updating the list as more things come to mind.

## DO NOT'S
- work **from your bedroom**: as well as being bad for your posture, one one hand it may be very tempting to sleep / watch your favourite show, on the other it may be tempting to work until very late.
- work **close to your loved ones**: there are some exceptions, but most people don't get that working remotely still means working(!!), and you will be interrupted far too often to be productive / concentrated
- **oblige to mobile notifications**: there is almost no such thing as an urgent notification, and **if you really must** or if you use it for work:
  - *use the web version* on your computer and not the mobile app
  - *do so in batches* - i.e. once every 1-2h during your break.
- **work with poor / unstable internet connection**: pretty self-explanatory, but often overlooked (e.g. working with mobile connection from a train).

## DO'S
### Productivity / Mind-tricks
- **separate work from play / sleep**: different environments, easier to concentrate, eventually your mind will automatically recognize when it's time to do either.
- **have a schedule**: remote working allows for flexibility, but do set yourself some guidelines (e.g. stop after 8h of work, try to free-up from other commitments when you are most productive)
- **recognize when you are productive**: (e.g. in the morning / late) everyone has his/her own rules / timelines for productivity. Use this to your advantage.
- **set small, achievable daily goals and weekly goals**: applicable also to non-remote workers, really
- **take few good breaks**: Taking too many breaks will probably be un-productive, but once you are isolated and concentrated, it's easy to go on an uninterrupted working streak... leading pretty soon to fatigue and slower problem-solving.
- **tackle complex problems with a fresh mind**: leave all simpler tasks for the later portion of your working day. If attempting to tackle a complex problem at the end of the day, best to leave for the following morning.

### Teamworking
- **allow for some overlap with teammates**: regardless of how much time difference between teammates quick catchup calls should be done at least once/twice a day.
- **written records of everything**: this is a must for remote teams, but there are strong benefits of doing this, regardless of being remote / co-located:
  - *daily updates* on what you achieved: helps keep track and provides team with knowledge (they will forget if it's just vocally mentioned in a standup)
  - *plan for the day/week*: gives clarity to who does it, helps you commit to the plan (less deviations / distractions)
  - *open in-progress PR's*: shows what you are doing. Your teammates can look at them too.
- **check for github notifications**: often, although not obsessively.
- **participate in the rest of the team's PR's**: get knowledge on the codebase
